### ISSUE TITLE
[BUG] S3 buckets Collection Should Be Iterable

### ISSUE THREAD

**Issue by stevenjackson121** [vemel/mypy_boto3_builder#7]
**Describe the bug**
`mypy_boto3_s3.S3ServiceResource.buckets` should support `__iter__` (and the return type of `__iter__` should be `mypy_boto3_s3.service_resource.Bucket`

I can't seem to reproduce this with the `mypy_boto3_s3.service_resource.Bucket.objects` collection, although they seem to be defined identically, so I'm not sure why `__iter__` seems to be supported for one but not the other.

**To Reproduce**
1. Install 'boto3-stubs[s3]'
2.  Run mypy on the following code sample

```python
import boto3
from mypy_boto3_s3 import S3ServiceResource

service_resource: S3ServiceResource = boto3.resource("s3")

buckets = service_resource.buckets.all()

for bucket in buckets:
    reveal_type(bucket)
```
3. Actual mypy output

```
s3_service_resource.py:8: error: "Type[ServiceResourceBucketsCollection]" has no attribute "__iter__" (not iterable)
s3_service_resource.py:9: note: Revealed type is 'Any'
Found 1 error in 1 file (checked 1 source file)
```

**Expected mypy-output**

```
No Errors (checked 1 source file)
```

**Additional context**
Mac OS, pipenv installation, `boto3` version 1.12.12, etc.

**Comment by stevenjackson121** [comment]
On a related note:

```
import boto3
from mypy_boto3_s3 import S3ServiceResource
from mypy_boto3_s3.service_resource import Bucket

service_resource: S3ServiceResource = boto3.resource("s3")

buckets = service_resource.buckets.pages()

for bucket in buckets:
    for m in bucket.multipart_uploads:
        reveal_type(m)
    for object in bucket.objects:
        reveal_type(object)
    reveal_type(bucket)
```
gives the following `mypy` output:
```
s3_service_resource.py:11: note: Revealed type is 'Any'
s3_service_resource.py:13: note: Revealed type is 'Any'
s3_service_resource.py:14: note: Revealed type is 'mypy_boto3_s3.service_resource.Bucket*'
```
Expected `mypy` output is 
```
s3_service_resource.py:11: note: Revealed type is 'mypy_boto3_s3.service_resource.MultiPartUpload'
s3_service_resource.py:13: note: Revealed type is 'mypy_boto3_s3.service_resource.Object'
s3_service_resource.py:14: note: Revealed type is 'mypy_boto3_s3.service_resource.Bucket*'
```

**Comment by stevenjackson121** [comment]
An entirely separate issue (I can create it separately if you like, but for DynamoDB you recommended collecting issues on the same Service in one Github Issue):

**Describe the bug**
`mypy_boto3_s3.S3ServiceResource.MultipartUpload` should support `Part()` (it should return a `mypy_boto3_s3.S3ServiceResource.MultipartUploadPart`, and is the only way to create these AFAIK

**To Reproduce**
Steps to reproduce the behavior:
1. Install 'boto3-stubs[s3]'
2.  Run mypy on the following code sample

```python
import boto3
from mypy_boto3_s3 import S3ServiceResource
from mypy_boto3_s3.service_resource import Bucket

service_resource: S3ServiceResource = boto3.resource("s3")

multipart_upload = service_resource.MultipartUpload("foo", "bar", "baz")
multipart_upload_part = multipart_upload.Part("1")
multipart_upload.abort()
```
3. Actual mypy output

```
s3_service_resource.py:8: error: "MultipartUpload" has no attribute "Part"
Found 1 error in 1 file (checked 1 source file)
```

**Expected mypy-output**

```
No Errors (checked 1 source file)
```

**Additional context**
Mac OS, pipenv installation, `boto3` version 1.12.12, etc.

**Comment by vemel** [comment]
Looks like collection-related types are incorrect. It is a bit hard to parse them from botocore shapes, so there are some hardcoded methods. Thank you for your report, I will update e2e tests and try to fix collections.

**Comment by vemel** [comment]
Fixes for first two issues are in master. Regarding the third one - looks like I am not adding sub-resources for collections https://boto3.amazonaws.com/v1/documentation/api/1.12.5/reference/services/s3.html#S3.MultipartUpload

**Comment by vemel** [comment]
Fixed both issues in new release https://github.com/vemel/mypy_boto3_builder/releases/tag/1.0.4-release . Please update `boto3-stubs` and `boto3`.

Let me know if it works as it should. Thank you!

**Comment by stevenjackson121** [comment]
Everything works. Thanks for the very quick response + fix!
